### PR TITLE
feat: disable request logging

### PR DESCRIPTION
**Pull request by SamSalvatico**
feat for issue: [#5409](https://github.com/fastify/fastify/issues/5409)

Check for the `disableRequestLogging` to be `false` or `undefined` before logging an error or a 404 during the request

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by metcoder95 — COMMENTED** [review]
Nice work!
Tho, I'm over the fence of considering this as a fix as was never stated that this was supported; I'd rather consider it more an enhancement.

Sadly tests are failing.

**Review by metcoder95 — APPROVED** [review]
LGTM 👍

**Comment by metcoder95** [comment]
Forgot to mention, can we add some documentation of this within the `disableRequestLogging` option documentation?

**Review by mcollina — APPROVED** [review]
lgtm

**Review by Uzlopak — APPROVED** [review]
hey ho lets go.

File: docs/Reference/Server.md [hunk 1]
```diff
@@ -385,9 +385,15 @@ been sent. By setting this option to `true`, these log messages will be
...
 
-Please note that this option will also disable an error log written by the
-default `onResponse` hook on reply callback errors. Other log messages 
-emitted by Fastify will stay enabled, like deprecation warnings and messages
+The other log entries that will be disabled are:
+- an error log written by the default `onResponse` hook on reply callback errors
+- the error and info logs written by the `defaultErrorHandler` 
+on error management
+- the info log written by the `fourOhFour` handler when a 
+not existent route is requested
```
Comment (Uzlopak) [inline docs/Reference/Server.md:393]: wait... a little typo
```suggestion
non existent route is requested
```

**Review by Uzlopak — CHANGES_REQUESTED** [review]
small nit

File: docs/Reference/Server.md (hunk 1 above)
Comment (Uzlopak) [reply docs/Reference/Server.md:393]: or 

```suggestion
not existing route is requested
```

**Review by Uzlopak — APPROVED** [review]
lets go

**Pull request merged**